src/cocotb/types/_logic_array.py

```diff
@@ -849,3 +860,9 @@ def resolve(self, resolver: ResolverLiteral) -> "LogicArray":
             TypeError: Unsupported *value* type.
         """
         return LogicArray(get_str_resolver(resolver)(str(self)), self.range)
+
+    def __copy__(self) -> "LogicArray":
+        return LogicArray(self._get_str(), self._range)
```

Comment:
This actually violates the definition of `copy.copy`. We would have to add some flags to force only `_value_as_array` to be maintained when the value is copied. That's ugly and awful, so I'm tempted to remove support for this operation.

Reply:
I did remove it.